core/internal/cltest/cltest.go

```diff
@@ -547,17 +537,19 @@ func (ta *TestApplication) Start() error {
 		return err
 	}
 
-	return ta.ChainlinkApplication.Start()
+	err = ta.ChainlinkApplication.Start()
+	if err != nil {
+		return err
+	}
+	ta.t.Cleanup(func() { require.NoError(ta.t, ta.Stop()) })
```

Comment:
Notice the old cleanup func at 460 was calling `StopIfStarted()` which is not overridden by `TestApplication`, so only the `ChainlinkApplication` was stopped. Solely using this for cleanup leaks `TestApplication` resources `Store`,`Server`, and `wsServer` which are cleaned up in the overridden `Stop()`.

Reply:
GREAT catch

I knew there was some broken logic hiding in there somewhere